src/hello_imgui/impl/runner_params.cpp

```diff
@@ -32,44 +32,41 @@ namespace HelloImGui
             auto _stringToSaneFilename=[](const std::string& s, const std::string& extension) -> std::string
             {
                 std::string filenameSanitized;
+
                 for (char c : s)
+                {
                     if (isalnum(c))
                         filenameSanitized += c;
                     else
                         filenameSanitized += "_";
+                }
+
                 filenameSanitized += extension;
                 return filenameSanitized;
             };
 
             if (! runnerParams.iniFilename.empty())
                 return runnerParams.iniFilename;
-            else
-            {
-                if (runnerParams.iniFilename_useAppWindowTitle && !runnerParams.appWindowParams.windowTitle.empty())
-                {
-                    std::string iniFilenameSanitized = _stringToSaneFilename(runnerParams.appWindowParams.windowTitle, ".ini");
-                    return iniFilenameSanitized;
-                }
-                else
-                    return "imgui.ini";
-            }
+
+            if (runnerParams.iniFilename_useAppWindowTitle && !runnerParams.appWindowParams.windowTitle.empty())
+                return _stringToSaneFilename(runnerParams.appWindowParams.windowTitle, ".ini");
+
+            return "imgui.ini";
+
         };
 
         auto mkdirToFilename = [](const std::string& filename) -> bool
         {
             std::filesystem::path p(filename);
             std::filesystem::path dir = p.parent_path();
+
             if (dir.empty())
                 return true;
+
             if (std::filesystem::exists(dir))
-            {
-                if (std::filesystem::is_directory(dir) || std::filesystem::is_symlink(dir))
-                    return true;
-                else
-                    return false;
-            }
-            else
```

Comment:
I sometimes use constructs like this make it easier inspect the behavior while I'm tuning the code and making sure it does what I want. 

But you are right: once validated, this can be removed